python/pyarrow/tests/test_compute.py

```diff
@@ -1637,6 +1637,48 @@ def test_is_null():
     assert result.equals(expected)
 
 
+def test_is_null_union():
```

Comment:
Is it necessary to add these tests on the Python side? You have similar tests in C++ already.